tools/docker/build.sh

```diff
@@ -6,24 +6,21 @@
 # See LICENSE file for more details.
 ###############################################################
 
-scriptdir="$(cd "${0%/*}" || exit 1; pwd)"
```

Comment:
To make this script source-able, it would be sufficient to change this to
```
scriptdir="$(dirname "${BASH_SOURCE[0]}")"
```
and change nothing else in the script.

It might still make sense to move the definition of ROOT_DIR into functions.sh, but I think it has only limited merit.